### PR TITLE
VB-3189 Bookings to review listing page

### PR DESCRIPTION
**Pull request by tpmcgowan**
* Listing page for bookings that need to be reviewed (action links present but linked pages not yet built)
* Supporting so far non-associations, prisoner release and visit type change notification events - and the different rendering requirements for each
* Bulleted list of checks is tied to the notification events supported by the API
* Integration tests
* Managed by `FEATURE_REVIEW_BOOKINGS` flag; limited to dev environment currently

Screenshot 2023-11-19 at 19 11 49

### CODE REVIEW

File: server/@types/bapv.d.ts [hunk 1]
```diff
@@ -143,3 +144,11 @@ export type VisitsPageSlot = {
 }
 
 export type FlashData = Record<string, string[] | Record<string, string | string[]>[]>
+
+export type VisitsReviewListItem = {
```
Comment (tpmcgowan) [inline server/@types/bapv.d.ts:148]: Represents a row in the bookings review list

File: server/constants/notificationEventTypes.ts
Comment (tpmcgowan) [inline server/constants/notificationEventTypes.ts:1]: Various lookups for notification types; tied to `NotificationType` `enum` in the API specification to make sure there are always entries for API-supported types.

File: server/services/visitNotificationsService.ts [hunk 2]
```diff
@@ -0,0 +1,47 @@
+import { VisitsReviewListItem } from '../@types/bapv'
+import { HmppsAuthClient, OrchestrationApiClient, RestClientBuilder } from '../data'
+import { NotificationCount } from '../data/orchestrationApiTypes'
+import { prisonerDateTimePretty } from '../utils/utils'
+
+export default class VisitNotificationsService {
```
Comment (tpmcgowan) [inline server/services/visitNotificationsService.ts:6]: Created new service to deal with visit notifications; should be a better fit as the journey is built out.

**Review by psoleckimoj — APPROVED** [review]
lgtm!

**Pull request merged**